### PR TITLE
FAI-13273: Add custom fields and populate tags field for TestRail

### PR DESCRIPTION
**Pull request by alexshifrin**
## Description

Added `custom_automation_type` and `custom_update_automation` fields and populated into Case tags.

## Type of change
- [ ] Bug fix
- [x] New feature
- [ ] Breaking change

https://faros-ai.atlassian.net/browse/FAI-13273

### CODE REVIEW

File: destinations/airbyte-faros-destination/src/converters/testrails/cases.ts [hunk 1]
```diff
@@ -22,15 +22,25 @@ export class Cases extends TestRailsConverter {
       testCase.suite_id,
       testCase.id
     );
-    const milestoneTag = `milestone:${testCase.milestone}`;
+    const milestoneTag = typeof testCase.milestone !== 'undefined'
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/testrails/cases.ts:25]: nit: `const milestoneTag = testCase.milestone ? `milestone:${testCase.milestone}` : null;`

Same for others

File: destinations/airbyte-faros-destination/src/converters/testrails/cases.ts [hunk 2]
```diff
@@ -22,15 +22,25 @@ export class Cases extends TestRailsConverter {
...
+      ? `milestone:${testCase.milestone}`
+      : null;
+    const automationTypeTag = typeof testCase.custom_automation_type !== 'undefined'
+      ? `automation_type:${testCase.custom_automation_type}`
+      : null;
+    const updateAutomationTag = typeof testCase.custom_update_automation !== 'undefined'
+      ? `update_automation:${testCase.custom_update_automation}`
+      : null;
+    const tags = [milestoneTag, automationTypeTag, updateAutomationTag].filter(tag => tag !== null);
+    const formattedTags = tags.map(tag => `{${tag}}`).join('');
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/testrails/cases.ts:35]: `tags` is an array of string in the model, I think we need to drop the `join` here

File: sources/testrails-source/src/testrails/models.ts [hunk 3]
```diff
@@ -57,6 +57,9 @@ export interface Case {
   readonly custom_preconds: string;
   readonly custom_steps: string;
   readonly custom_expected: string;
+  readonly custom_automation_type?: number;
+  readonly custom_update_automation?: boolean;
+  readonly custom_notes?: string;
```
Comment (ypc-faros) [inline sources/testrails-source/src/testrails/models.ts:62]: You added this but it's not used in the converter...

**Comment by ypc-faros** [comment]
@alexshifrin do this: https://github.com/faros-ai/airbyte-connectors/pull/1707#issuecomment-2384597183

**Comment by alexshifrin** [comment]
I have read the CLA Document and I hereby sign the CLA

File: destinations/airbyte-faros-destination/src/converters/testrails/cases.ts (hunk 1 above)
Comment (alexshifrin) [reply destinations/airbyte-faros-destination/src/converters/testrails/cases.ts]: Updated to use lodash isNil

File: sources/testrails-source/src/testrails/models.ts (hunk 3 above)
Comment (alexshifrin) [reply sources/testrails-source/src/testrails/models.ts:62]: Included the custom_notes to be used for the Test Case description.

File: destinations/airbyte-faros-destination/src/converters/testrails/cases.ts [hunk 4]
```diff
@@ -22,15 +24,25 @@ export class Cases extends TestRailsConverter {
...
+      ? `update_automation:${testCase.custom_update_automation}`
+      : null;
+    const tags = [milestoneTag, automationTypeTag, updateAutomationTag].filter(tag => tag !== null);
 
     res.push({
       model: 'qa_TestCase',
       record: {
         uid,
         name: testCase.title,
+        description: testCase.custom_notes,
```
Comment (ypc-faros) [inline destinations/airbyte-faros-destination/src/converters/testrails/cases.ts:43]: Use `Utils.cleanAndTruncate(testCase.custom_notes)`

See example here: https://github.com/faros-ai/airbyte-connectors/blob/94da4a7038219335814005b7b284ecc8ce042c10/destinations/airbyte-faros-destination/src/converters/asana/tasks.ts#L146

You can omit the second param, it uses 1000 by [default](https://github.com/faros-ai/faros-js-client/blob/221089d74b041d389abdd5f82436d4444979878b/src/utils.ts#L151)

**Review by ypc-faros — APPROVED** [review]

File: destinations/airbyte-faros-destination/src/converters/testrails/cases.ts [hunk 5]
```diff
@@ -22,15 +25,25 @@ export class Cases extends TestRailsConverter {
...
+    const milestoneTag = !isNil(testCase.milestone)
+      ? `milestone:${testCase.milestone}`
+      : null;
+    const automationTypeTag = !isNil(testCase.custom_automation_type)
+      ? `automation_type:${testCase.custom_automation_type}`
+      : null;
+    const updateAutomationTag = !isNil(testCase.custom_update_automation)
+      ? `update_automation:${testCase.custom_update_automation}`
+      : null;
+    const tags = [milestoneTag, automationTypeTag, updateAutomationTag].filter(tag => tag !== null);
```
Comment (tovbinm) [inline destinations/airbyte-faros-destination/src/converters/testrails/cases.ts:37]: Nice.

**Pull request merged**